docs/index.md

```diff
@@ -459,6 +459,531 @@ class Settings(BaseSettings):
     model_config = SettingsConfigDict(env_file='.env', extra='ignore')
 ```
 
+## Command Line Support
+
+Pydantic settings provides integrated CLI support, making it easy to quickly define CLI applications using Pydantic
+models. There are two primary use cases for Pydantic settings CLI:
+
+1. When using a CLI to override fields in Pydantic models.
+2. When using Pydantic models to define CLIs.
+
+By default, the experience is tailored towards use case #1 and builds on the foundations established in [parsing
+environment variables](#parsing-environment-variables). If your use case primarily falls into #2, you will likely want
+to enable [enforcing required arguments at the CLI](#enforce-required-arguments-at-cli).
+
+### The Basics
+
+To get started, let's look at a basic example for defining a Pydantic settings CLI:
+
+```py test="skip"
```

Comment:
This example isn't very "basic", better to start with a trivial example.

Reply:
also, would be good if we didn't have to skip tests for most of these.

Reply:
> This example isn't very "basic", better to start with a trivial example.

Sure, will update.

> also, would be good if we didn't have to skip tests for most of these.

Agreed. I tried to keep them where possible. The issue we encountered was for doc examples showing CLI `--help` which result in an argparse `SystemExit` causing pytest to fail. If there is a way to have the doc tests handle `SystemExit`, we could re-enable these.

Reply:
@hramezani @samuelcolvin 

I think we're at an impasse for the doc related tests on the `--help` output. I updated the examples to use a custom CLI source so the help text could be directly grabbed from the root parser. While somewhat cringey, it did allow for a real doc test on the `--help` output.

However, I had forgotten Python 3.9 -> 3.10 `argparse` changed the help text slightly, meaning we would also have to update the examples to account for the difference. This is ok in the [unittests](https://github.com/pydantic/pydantic-settings/blob/9679c107daf910ffca6ea769b4d1ed1eb47e1024/tests/test_settings.py#L2486), but for the help docs it will just further obfuscate the examples. Therefore, looking for guidance on how to best proceed given the below options:

1. Use pytest="skip" again for doc tests related to `--help` output?
2. Update the docs to break up the python code from the expected help text?
3. Other ideas?

Reply:
I skipped them as it is cleanest from docs perspective.